aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -215,7 +207,7 @@ def get_options(self):
         :return: dictionary of the options and their values
         """
         options = {}
-        for name in self.options.keys():
+        for name in self.process_class.spec_options.keys():
```

Comment:
Will this introduce an actual change in the behavior of the method?
Since `get_option()` will still act on the node, the extra keys that will be found through this will always be None from the node options.
Furthermore, if one should change the process class spec, removing a key, I guess this method, `get_options`, won't actually find all options set for this node. Unless your intention is to ignore "legacy" options for a node, even though they exist?

Reply:
It does *not* change the behavior of the method before this change. However, you are right, if the process spec of the process class is changed and for example removes an option, then this will no longer find it. The problem is however, that "options" are simply stored as attributes and just from their key in the attributes it is impossible to determine whether it is actually an option or a random other attribute. The only solution would be to put a requirement on the key of option attributes and say for example they start with `option_` prefix. However, this would require a data migration that is far from easy. For example, how do you determine for the currently existing attributes which are options and which are not?

Reply:
> ... The only solution would be to put a requirement on the key of option attributes and say for example they start with `option_` prefix. ...

Or they could be separated into a single `options` attribute dict.

> ... However, this would require a data migration that is far from easy. For example, how do you determine for the currently existing attributes which are options and which are not?

Yes. Yes, this is not something that seems like an urgent thing at all to do, nor even something that is necessary I think?
If I should shoot in the dark I would suggest going through the attributes based on the process class spec, but this will not necessarily get all option items in the end. Yeah --- let's not :)

Reply:
So we'll leave it as it is for now